mmeds/database.py

```diff
@@ -38,29 +38,27 @@ def upload_metadata(args):
     p = MetaDataUploader(subject_metadata, specimen_metadata, owner, 'qiime', reads_type,
                          barcodes_type, study_name, False, datafiles,
                          False, True, access_code)
-    p.start()
-    p.join()
-    return p.exitcode
+    p.run()
+    return 0
 
 
 def upload_otu(args):
     (subject_metadata, specimen_metadata, path, owner, study_name, otu_table, access_code) = args
     datafiles = {'otu_table': otu_table}
     p = MetaDataUploader(subject_metadata, specimen_metadata, owner, 'sparcc', 'otu_table',
                          None, study_name, False, datafiles, False, True, access_code)
-    p.start()
-    p.join()
-    return p.exitcode
+    p.run()
+    return 0
 
 
 def upload_lefse(args):
     (subject_metadata, specimen_metadata, path, owner, study_name, lefse_table, access_code) = args
-    datafiles = {'lefse_table:': lefse_table}
+    datafiles = {'lefse_table': lefse_table}
+
     p = MetaDataUploader(subject_metadata, specimen_metadata, owner, 'lefse', 'lefse_table',
                          None, study_name, False, datafiles, False, True, access_code)
-    p.start()
-    p.join()
-    return p.exitcode
+    p.run()
+    return 0
```

Comment:
why return 0 intstead of the exitcode?

Reply:
I can see later in the PR that 0 is an exitcode, but why not return p.exitcode?

Reply:
because `p.run()` doesn't actually spin up a new process unlike `p.start()` so  `p.exitcode == None`. I changed it to this, partially for efficiency and partially to make debugging easier. These functions only get used during testing and were waiting for the processes to finish before returning anyway

Reply:
ah gotcha